controllers/databaseclusterbackup_controller.go

```diff
@@ -153,25 +160,206 @@ func (r *DatabaseClusterBackupReconciler) SetupWithManager(mgr ctrl.Manager) err
 	if err == nil {
 		if err := r.addPXCToScheme(r.Scheme); err == nil {
 			controller.Owns(&pxcv1.PerconaXtraDBClusterBackup{})
+			controller.Watches(
+				&pxcv1.PerconaXtraDBClusterBackup{},
+				handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, obj client.Object) []reconcile.Request {
+					return r.tryCreateDBBackups(ctx, obj, r.tryCreatePXC)
+				}))
 		}
 	}
 
 	err = r.Get(ctx, types.NamespacedName{Name: psmdbBackupCRDName}, unstructuredResource)
 	if err == nil {
 		if err := r.addPSMDBToScheme(r.Scheme); err == nil {
 			controller.Owns(&psmdbv1.PerconaServerMongoDBBackup{})
+			controller.Watches(
+				&psmdbv1.PerconaServerMongoDBBackup{},
+				handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, obj client.Object) []reconcile.Request {
+					return r.tryCreateDBBackups(ctx, obj, r.tryCreatePSMDB)
+				}))
 		}
 	}
 
 	err = r.Get(ctx, types.NamespacedName{Name: pgBackupCRDName}, unstructuredResource)
 	if err == nil {
 		if err := r.addPGToScheme(r.Scheme); err == nil {
 			controller.Owns(&pgv2.PerconaPGBackup{})
+			controller.Watches(
+				&pgv2.PerconaPGBackup{},
+				handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, obj client.Object) []reconcile.Request {
+					return r.tryCreateDBBackups(ctx, obj, r.tryCreatePG)
+				}))
 		}
 	}
 	return controller.Complete(r)
 }
 
+func (r *DatabaseClusterBackupReconciler) tryCreateDBBackups(
+	ctx context.Context,
+	obj client.Object,
+	createBackupFunc func(ctx context.Context, obj client.Object) error,
+) []reconcile.Request {
+	logger := log.FromContext(ctx)
+	if len(obj.GetOwnerReferences()) == 0 {
+		err := createBackupFunc(ctx, obj)
+		if err != nil {
+			logger.Error(err, fmt.Sprintf("Failed to create DatabaseClusterBackup %s", obj.GetName()))
+		}
+	}
+
+	return []reconcile.Request{{
```

Comment:
I think this request may need to be moved inside the `if` loop if there's no error, right?

Reply:
No, because it defines the objects to be reconciled (as discussed above in [the thread](https://github.com/percona/everest-operator/pull/42#discussion_r1303955597)). Even if we don't need to create a new DatabaseBackup, we still need to reconcile it so it reflects the upstream backup status.